pyswmm/nodes.py

```diff
@@ -0,0 +1,626 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014 Bryant E. McDonnell
+#
+# Licensed under the terms of the BSD2 License
+# See LICENSE.txt for details
+# -----------------------------------------------------------------------------
+"""Nodes module for the pythonic interface to SWMM5."""
+
+from toolkitapi import *
+from swmm5 import SWMMException, PYSWMMException
+
+class Nodes(object):
+    """
+    Node Iterator Methods
+
+    :param object model: Open Model Instance 
+
+    Examples:
+        
+    >>> from pyswmm import Simulation
+    >>>
+    >>> with Simulation('../test/TestModel1_weirSetting.inp') as sim:
+    ...     for node in Nodes(sim):
+    ...         print node
+    ...         print node.nodeid
+    ...
+    >>> <swmm5.Node object at 0x031B0350>
+    >>> J1
+    >>> <swmm5.Node object at 0x030693D0>
+    >>> J2
+    >>> <swmm5.Node object at 0x031B0350>
+    >>> J3
+    >>> <swmm5.Node object at 0x030693D0>
+    >>> J0
+
+    Iterating or Nodes Object
+    
+    >>> nodes = Nodes(sim)
+    >>> for node in nodes:
+    ...     print node.nodeid
+    >>> J0
+    >>> J1
+    >>> J2
+    >>> J3
+    
+    Testing Existence
+    
+    >>> nodes = Nodes(sim)
+    >>> "J1" in nodes
+    >>> True
+
+    Initializing a node Object
+
+    >>> nodes = Nodes(sim)
+    >>> j1 = nodes['J1']
+    >>> print(j1.invert_elevation)
+    >>> 12
+    >>>
+    >>> j1.invert_elevation = 200
+    >>> print(j1.invert_elevation)
+    >>> 200
+
+    
+    """
+    def __init__(self, model):
+        if not model._model.fileLoaded:
+            raise PYSWMMException("SWMM Model Not Open")
+        self._model = model._model
+        self._cuindex = 0
+        self._nNodes = self._model.getProjectSize(ObjectType.NODE)
+        
+    def __len__(self):
+        """Return number of nodes. Use the expression 'len(Nodes)'.
+
+        :return: Number of Nodes
+        :rtype: int
+        
+        """
+        return self._nNodes
+
+    def __contains__(self, nodeid):
+        """Checks if Node ID exists
+
+        :return: ID Exists
+        :rtype: bool        
+        """
+        return self._model.ObjectIDexist(ObjectType.NODE, nodeid)
+
+    def __getitem__(self, nodeid):
+        if self.__contains__(nodeid):
+            return Node(self._model, nodeid)
+        else:
+            raise PYSWMMException("Node ID Does not Exist")
+        
+    def __iter__(self):
+        return self
+    
+    def next(self):
+        if self._cuindex < self._nNodes:
+            nodeobject = Node(self._model, self.nodeid)
+            self._cuindex+=1 #Next Iteration
+            return nodeobject
+        else:
+            raise StopIteration()
+        
+    @property
+    def nodeid(self):
```

Comment:
`node_` inside node object is redundant but `id` is a reserved word so many `nid` otherwise we would need to use `id_`, which many libraries do?

Why is there a nodeid in the nodes objetct?